### PR TITLE
bpo-36707: Document "m" removed from sys.abiflags

### PR DESCRIPTION
**Pull request by vstinner**
https://bugs.python.org/issue36707

### CODE REVIEW

**Comment by vstinner** [comment]
@hroncok @ned-deily: Would you mind to review this doc change?

File: Doc/whatsnew/3.8.rst [hunk 1]
```diff
@@ -955,6 +955,20 @@ Optimizations
...
+  Example of changes:
+
+  * Only ``python3.8`` program is installed, ``python3.8m`` program is gone.
+  * Only ``python3.8-config`` script is installed, no ``python3.8m-config`` is
+    gone.
+  * The ``m`` flag has been removed from the suffix of dynamic library
+    filenames. On Linux, for example, the Python 3.7 suffix
+    ``.cpython-37m-x86_64-linux-gnu.so`` became
+    ``.cpython-38-x86_64-linux-gnu.so`` in Python 3.8.
+
```
Comment (ned-deily) [inline Doc/whatsnew/3.8.rst:973]: It should also be noted that this affects extension modules in the standard library as well as those produced and installed by third-party packages, like those downloaded from PyPI.

Comment (vstinner) [reply Doc/whatsnew/3.8.rst:973]: I updated my PR. Does it look better now?

**Comment by vstinner** [comment]
I rebased my PR to fix a conflict.

**Review by ned-deily — APPROVED** [review]
My preference remains to not make the flags change at all but, if we must, this documents it adequately, I think.  :). Thanks!

File: Doc/whatsnew/3.8.rst [hunk 2]
```diff
@@ -955,6 +955,22 @@ Optimizations
...
 
+* Default :data:`sys.abiflags` became an empty string: the ``m`` flag for
+  pymalloc became useless (builds with and without pymalloc are ABI compatible)
+  and so has been removed. (Contributed by Victor Stinner in :issue:`36707`.)
+
+  Example of changes:
+
+  * Only ``python3.8`` program is installed, ``python3.8m`` program is gone.
+  * Only ``python3.8-config`` script is installed, no ``python3.8m-config`` is
+    gone.
```
Comment (yan12125) [inline Doc/whatsnew/3.8.rst]: > no ``python3.8m-config`` is gone.

Should be?

> ``python3.8m-config`` is gone.

**Comment by vstinner** [comment]
> My preference remains to not make the flags change at all

Well, "m" in ABI flags became useless around Python 3.4. I don't see the point of pretending that the "m" ABI is different and then provide two names for the same thing:

* python3.8 **and** python3.8m
* python3.8-config **and** python3.8m-config
* man python3.8 **and** man python3.8m
* etc.

I'm not sure why, but on Fedora /usr/bin/python3.7 and /usr/bin/python3.7m are two separated files... with the same content (ex: same MD5 sum). Well, it's just 17 KiB each :-) /usr/bin/python3.7-config is a symlink to python3.7m-config.

The change itself is painful for distributors like Red Hat (Fedora, RHEL), but things should be simpler once the change is done.

My long-term goal is to get a single ABI for everything: CPython, PyPy, RustPython, MicroPython, etc. :-) Maybe it's not possible, but at least I would like to move towards this goal!

Overall project: https://pythoncapi.readthedocs.io/

**Comment by vstinner** [comment]
Thanks @ned-deily and @yan12125 for the review ;-)

**Review by hroncok — COMMENTED** [review]
looks good, sorry for not responding faster

**Pull request merged**